pkg/kpipelines/pipelines.go

```diff
@@ -214,3 +222,554 @@ func PipelineResourceName(gitInfo *gits.GitRepository, branch string, context st
 	resourceName := kube.ToValidNameTruncated(dirtyName, 35)
 	return resourceName
 }
+
+var randReader = rand.Reader
+
+func scopedEnv(newEnv []v1.PipelineStructureEnvVar, parentEnv []corev1.EnvVar) []corev1.EnvVar {
```

Comment:
Any particular reason to move this stuff from its previous location? I don't feel too strongly about it, but I liked that `syntax/pipeline.go` had nothing to do with client-go related code or updating CRDs in the cluster or anything, whereas now this file now has both the pure translation logic and non-pure CRD creation/update logic (although they don't interact). `step_create_task.go` also has some translation logic as well, so we don't really have a clear separation between the files in general. Not really something to fix now but would be good to be thinking about where we can draw boundaries to keep the code easy to maintain.

Reply:
Separation of concerns - `types_pipelinestructure.go` is just the `PipelineStructure` CRD, its validation, and methods on its types, while `pipelines.go` has all the translation. It felt silly to have a separate package/file for just the translation when we already had this package/file here we could move it into. And we will be rewriting the non-pure-translation stuff soon to go build pack->`PipelineStructure`->`Pipeline` et al, so splitting that up just for now doesn't make a lot of sense to me.